### PR TITLE
Ignoring duplicated cipher suites

### PR DESCRIPTION
**Pull request by shonecyx**
Fix https://github.com/istio/istio/issues/36805

**To help us figure out who should review this PR, please put an X in all the areas that this PR affects.**

- [ ] Configuration Infrastructure
- [ ] Docs
- [ ] Installation
- [ X] Networking
- [ ] Performance and Scalability
- [ ] Policies and Telemetry
- [ ] Security
- [ ] Test and Release
- [ ] User Experience
- [ ] Developer Infrastructure

### CODE REVIEW

**Review by Arhell — COMMENTED** [review]
/ok-to-test

**Review by hzxuzhonghu — COMMENTED** [review]
It's better also to give some warning to users in validation stage

**Comment by zirain** [comment]
> It's better also to give some warning to users in validation stage

see https://github.com/istio/istio/pull/34109

**Comment by shonecyx** [comment]
> > It's better also to give some warning to users in validation stage
> 
> see #34109

Add validation, thanks. cc @hzxuzhonghu  @zirain

File: pilot/pkg/networking/core/v1alpha3/gateway.go [hunk 1]
```diff
@@ -1051,13 +1052,13 @@ func buildGatewayVirtualHostDomains(hostname string, port int) []string {
...
-	ret := make([]string, 0, len(suites))
+	ret := sets.NewSet()
 	for _, s := range suites {
-		if security.IsValidCipherSuite(s) {
-			ret = append(ret, s)
+		if security.IsValidCipherSuite(s) && !ret.Contains(s) {
+			ret = ret.Insert(s)
 		} else if log.DebugEnabled() {
-			log.Debugf("ignoring unsupported cipherSuite: %q for server %s", s, server.String())
+			log.Debugf("ignoring unsupported/duplicated cipherSuite: %q for server %s", s, server.String())
```
Comment (howardjohn) [inline pilot/pkg/networking/core/v1alpha3/gateway.go:1060]: can we have a different error for each case? May cause confusion as is

File: pilot/pkg/networking/core/v1alpha3/gateway.go [hunk 2]
```diff
@@ -1051,13 +1052,13 @@ func buildGatewayVirtualHostDomains(hostname string, port int) []string {
...
 		}
 	}
-	return ret
+	return ret.SortedList()
```
Comment (howardjohn) [inline pilot/pkg/networking/core/v1alpha3/gateway.go:1063]: This will break I think, `cipher_suites` order matters - it determines precedence.

**Review by howardjohn — CHANGES_REQUESTED** [review]
LGTM if we retain order

File: pilot/pkg/networking/core/v1alpha3/gateway.go (hunk 1 above)
Comment (shonecyx) [reply pilot/pkg/networking/core/v1alpha3/gateway.go]: Done

File: pilot/pkg/networking/core/v1alpha3/gateway.go (hunk 2 above)
Comment (shonecyx) [reply pilot/pkg/networking/core/v1alpha3/gateway.go]: Fixed, thanks

**Review by hzxuzhonghu — APPROVED** [review]

**Review by howardjohn — APPROVED** [review]

**Pull request merged**